### PR TITLE
Refactor auth to use AutoValue classes and implement login barrier

### PR DESCRIPTION
**Pull request by ptwu**
### Summary
This PR implements AutoValue data classes to represent `AuthInfo` objects, and adds a login barrier to the `my-trips.html` and `trips-network.html` pages. This is achieved through the `authReload()` function in `constant-script.js`, which is shared across all pages. However, the `authReload()` function is meant only to be called in pages that require login to access, and as such, the `onload` functions for each of these pages call it. 

This PR also changes the logout button to be its own discrete button instead of making it "Sign out with Google".

### Screenshot
Overnightly

### Test Plan
Run a local server of the site and try navigating to a authwalled page, like My Trips. See that it redirects you to login first. Login, and try logging out from the My Trips page. See that it works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,4 +6,6 @@ Overnightly is a webapp that makes overnight trip planning a snap.
 
 ### Commands:
 
-- `mvn package appengine:run` runs a local server with the webapp.
+- `mvn package appengine:run` runs a local server at port 8080 with the webapp.
+- `mvn jasmine:bdd` runs a live-reloading local server at port 8000 evaluating
+  the test cases for the frontend found at `src/main/webapp/tests/`.
```
Comment (mffabrigas) [inline README.md:11]: Good documentation!

**Comment by mffabrigas** [comment]
However, when logged out and on the home page for the first time, it may be better to leave the My Trips and Trips Network tabs invisible by default to explicitly direct the user to login first to use the website.

**Comment by ptwu** [comment]
> However, when logged out and on the home page for the first time, it may be better to leave the My Trips and Trips Network tabs invisible by default to explicitly direct the user to login first to use the website.

That's actually a great idea since after the user logs in, I redirect them to my.trips.html anyway. Going to change that right now.

**Comment by ptwu** [comment]
> LGTM. Good choice to manually check for user login with each page so no one can forcibly link to an auth-walled page, though in the future we may need to add a redirect message to tell users if they were logged out during use of the website and are being redirected to login so that transition isn't so jarring.

Yeah, I agree it can be a bit shocking, but I think Google does the same with Docs/Drive links that you don't have access to, so I based it off of that.

**Comment by mffabrigas** [comment]
> > LGTM. Good choice to manually check for user login with each page so no one can forcibly link to an auth-walled page, though in the future we may need to add a redirect message to tell users if they were logged out during use of the website and are being redirected to login so that transition isn't so jarring.
> 
> Yeah, I agree it can be a bit shocking, but I think Google does the same with Docs/Drive links that you don't have access to, so I based it off of that.

True, its not a priority, just a small detail we can maybe account for later if we have time.

**Pull request merged**